fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/DynamicRegistries.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.registry;
+
+import java.util.List;
+
+import com.mojang.serialization.Codec;
+import org.jetbrains.annotations.Unmodifiable;
+
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.registry.RegistryLoader;
+
+import net.fabricmc.fabric.impl.registry.sync.DynamicRegistriesImpl;
+
+/**
+ * Contains methods for registering and accessing dynamic registries.
+ *
+ * <p>TODO: Add usage example
+ */
+public final class DynamicRegistries {
+	private DynamicRegistries() {
+	}
+
+	/**
+	 * Returns an unmodifiable list of all dynamic registries, including modded ones.
+	 *
+	 * <p>The list will not reflect any changes caused by later registrations.
+	 *
+	 * @return an unmodifiable list of all dynamic registries
+	 */
+	public static @Unmodifiable List<RegistryLoader.Entry<?>> getDynamicRegistries() {
+		return DynamicRegistriesImpl.getDynamicRegistries();
+	}
+
+	/**
+	 * Registers a non-synced dynamic registry.
+	 *
+	 * <p>The entries of the registry will be loaded from data packs at the file path
```

Comment:
Question: what is the path for tags?

Reply:
Currently unadjusted from vanilla's unprefixed directory - I meant to make a PR to fix it separately (as it also affects static registries), but forgot... I can add it here too.